source/index.html.md

```diff
@@ -104,6 +104,153 @@ curl -X DELETE \
+
+### Verify device
+
+To verify device you need to provide a signature of OTP received on your mobile phone.
+
+```shell
+curl "https://api.kontist.com/api/user/devices/4e310a55-1b1a-4efb-b9a5-fd04491bdd21/verify" \
+  -H "Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l" \
+  -H "Content-Type: application/json" \
+  -X POST \
+  -d '{
+        "challengeId": "4e310a55-1b1a-4efb-b9a5-fd04491bdd21",
+        "signature": "30440220220B71BA03178A43B6CFA766F1B520CA1A626777F76B21253F9EC5039F4A0EB3022043CF2685C8F695F434862EADD1D5F5D6F68C29E875F755D058070A71E8338E11"
+      }'
+```
+
+> The above command returns `204 No Content` in case of success.
+
+#### HTTP Request
+
+`POST https://api.kontist.com/api/user/devices/{device_id}/verify`
+
+#### Request body
+
+| Parameter   | Mandatory | Description                                                                    |
+| ----------- | --------- | ------------------------------------------------------------------------------ |
+| challengeId | yes       | ID of the challenge recieved during device creation                            |
+| signature   | yes       | The hex-encoded, raw signature for the OTP recived in SMS                      |
+
+### Create device challenge
+
+After the device is created and verified, you need to create a device challenge. As a response, you will receive `stringToSign` that should be used during verification of device challenge.
+
+```shell
+curl "https://api.kontist.com/api/user/devices/4e310a55-1b1a-4efb-b9a5-fd04491bdd21/challenges" \
+  -H "Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l" \
+  -X POST
+```
+
+> The above command returns JSON structured like this:
+
+```json
+{
+  "id": "5f7c36e2-e0bf-4755-8376-ac6d0711192e",
+  "stringToSign": "9e2d45df-9b00-49d3-9064-29b86374fe67"
+}
+```
+
+#### HTTP Request
+
+`POST https://api.kontist.com/api/user/devices/{device_id}/challenges`
+
+#### Response
+| Field         | Description                                                                    |
+| ------------- | ------------------------------------------------------------------------------ |
+| id            | ID of the challenge                                                            |
+| stringToSign  | Challenge string that should be signed by device private key                   |
+
+
+### Verify device challenge
+
+To verify device challenge you need to provide a signature of `stringToSign` received after challenge creation.
+
+```shell
+curl "https://api.kontist.com/api/user/devices/4e310a55-1b1a-4efb-b9a5-fd04491bdd21/challenges/5f7c36e2-e0bf-4755-8376-ac6d0711192e/verify" \
+  -H "Authorization: Basic QWxhZGRpbjpPcGVuU2VzYW1l" \
+  -H "Content-Type: application/json" \
+  -X POST \
+  -d '{
+        "signature": "FF93DF062DB808E35AB9D28D80E0B261C7313C69785471954C05A474156BA7A7F07F2F0E7E805513754A8119BBF172E1E6D0103901249CE8DE012E5E61FDA36AD06405341043"
+      }'
+```
+
+> The above command returns JSON structured like this:
+
+```json
```

Comment:
Does it make sense to decode this token and give a hint how you can distinguish a confirmed from an unconfirmed token (cnf.kid)?

Reply:
I think it's not needed for API users to know about it. Let's leave it as implementation detail :)